gnovm/pkg/gnolang/op_exec.go

```diff
@@ -904,7 +904,7 @@ func (m *Machine) doOpSwitchClause() {
 	// caiv := m.PeekValue(2) // switch clause case index (reuse)
 	cliv := m.PeekValue(3) // switch clause index (reuse)
 	idx := cliv.GetInt()
-	if idx >= len(ss.Clauses) {
+	if int(idx) >= len(ss.Clauses) { // XXX: idx may overflow on 32 bits platforms.
```

Comment:
```suggestion
	if int(idx) >= len(ss.Clauses) {
```

If there are more than 2^31 switch cases, we have bigger problems.